src/hooks/usePDFExport.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useRef } from 'react';
+import { useReactToPrint } from 'react-to-print';
+import { IconFileDownload } from '@trussworks/react-uswds';
+import classNames from 'classnames';
+
+import IconButton from 'components/shared/IconButton';
+
+type UsePDFExportProps = {
+  /** PDF filename */
+  filename: string;
+  /** Optional title to display at top of PDF */
+  title?: string;
+};
+
+type PDFExportButtonProps = {
+  children: React.ReactNode;
+  disabled?: boolean;
+  className?: string;
+};
+
+/**
+ * Hook for using ReactToPrint export to PDF functionality
+ *
+ * Returns `PDFExportButton` and `PDFExportWrapper` components
+ */
+const usePDFExport = ({ filename, title }: UsePDFExportProps) => {
+  const printRef = useRef<HTMLDivElement>(null);
+
+  const handlePrint = useReactToPrint({
+    documentTitle: filename,
+    content: () => printRef.current,
+    // The lib default is to have no margin, which hides window.prints()'s built in pagination
+    // Set auto margins back to show everything the browser renders
+    pageStyle: `
+      @page {
+        margin: auto;
+      }
+    `
+  });
```

Comment:
I did a little more digging on using a hook to return components.  My understanding was that hooks should be used for extracting logic, not components.  I think these should probably be components in themselves.  I think this could probably opened up for more of a discussion, but my gut was that this isn't the intended use of hooks.

These a a few references discussing this topic
https://www.reddit.com/r/reactjs/comments/t7dg4e/custom_hook_returning_a_component_is_it_an/
This thread has links to different resources on the subject

https://stackoverflow.com/questions/75871491/is-it-a-good-idea-to-return-component-from-custom-hook-in-react

https://www.reddit.com/r/reactjs/comments/9yq1l8/how_do_you_feel_about_a_hook_returning_components/

Reply:
Using (react) features as they're intended will help with the overall readability of our code to other devs.

To me the biggest point in this case is about the separation of concerns. Since this piece in particular is only returning components, it is confusing that it's from a hook.

I did some hook anti-pattern with the RTE external links because of trying to rush out that patch for the sprint. It was quicker for me to figure out the patch with everything in one run around the same code block, but it could definitely use another round of simplification for more permanent code.

If there's no rush, after something is working, taking time to separate concerns and simplifying by following conventions would be worth it.

PS I know that the "New React Hooks Pattern? Return a Component" article offers a solution, which I haven't read through too carefully yet, but it seems like a different situation where the author came up with a whole pattern to generally handle another UI lib, where they're interested in reducing certain boilerplate.

Reply:
@patrickseguraoddball that makes sense - I'm good with not returning components from hooks. Thanks for sharing that information! I removed the hook and used the original `useReactToPrint` hook instead. I kept the `PDFExportButton` component but moved it into the `PDFExport` component file.

I also removed the turquoise border from the PDF titles.